File: exercises/week05/exercise-dp-two-scala/src/abstractfactory/LondonParserFactory.scala
package abstractfactory

class LondonParserFactory extends AbstractParserFactory {
  def getParserInstance(parserType: String): XMLParser = {
    parserType match {
      case "LondonERROR" =>
        return new LondonErrorXMLParser
      case "LondonFEEDBACK" =>
        return new LondonFeedbackXMLParser
      case "LondonORDER" =>
        return new LondonOrderXMLParser
      case "LondonRESPONSE" =>
        return new LondonResponseXMLParser
    }
  }
}

File: exercises/week05/exercise-dp-two/src/abstractfactory/LondonParserFactory.java
package abstractfactory;

public class LondonParserFactory implements AbstractParserFactory {

    @Override
    public XMLParser getParserInstance(String parserType) {

        switch (parserType) {
            case "LondonERROR":
                return new LondonErrorXMLParser();
            case "LondonFEEDBACK":
                return new LondonFeedbackXMLParser();
            case "LondonORDER":
                return new LondonOrderXMLParser();
            case "LondonRESPONSE":
                return new LondonResponseXMLParser();
        }

        return null;
    }

}
